src/IECoreImage/DisplayDriverServer.cpp

```diff
@@ -472,6 +473,20 @@ void DisplayDriverServer::Session::handleReadOpenParameters( const boost::system
 						);
 				m.mergeCount = sessionClientsData->readable();
 			}
+			if ( channelNames->readable() != m.mergeDriver->channelNames() )
+			{
+				throw IECore::InvalidArgumentException( "Merge driver has incompatible channel names." );
+			}
+			if ( displayWindow->readable() != m.mergeDriver->displayWindow() )
+			{
+				throw IECore::InvalidArgumentException( "Merge driver has incompatible display window." );
+			}
+			if ( ++m.usageCount > sessionClientsData->readable() )
+			{
+				throw IECore::InvalidArgumentException( "Number of merge clients (" + std::to_string( m.usageCount )
+						+ ") is more than expected amount (" + std::to_string( m.mergeCount ) + ")." );
+			}
```

Comment:
It seems unfortunate that we're testing against `sessionClientsData` from the _new_ driver here, since we have no guarantee that it has the same value that was used to initialise `m.mergeCount` in the first place. But we can't test against `m.mergeCount` either, since that is decremented in `imageClose`. Can we rejig our two state variables to make this a bit more watertight?